kafka-rest/src/main/java/io/confluent/kafkarest/DefaultKafkaRestContext.java

```diff
@@ -71,16 +80,34 @@ public synchronized Admin getAdmin() {
     return adminClient;
   }
 
+  @Override
+  public synchronized Producer<byte[], byte[]> getProducer() {
+    if (producer == null) {
+      producer =
```

Comment:
side commentary: i don't like implicit `this`, but languages fault, and this seems to be the convention in this codebase.